packages/rps/src/core/app-data.ts

```diff
@@ -1,26 +1,73 @@
 import {Weapon} from './weapons';
+import {BigNumber, defaultAbiCoder, bigNumberify, keccak256} from 'ethers/utils';
+import {HashZero} from 'ethers/constants';
+import {randomHex} from '../utils/randomHex';
 
-export interface Resting {
-  type: 'resting';
+export enum PositionType {
+  Start, // 0
+  RoundProposed, // 1
+  RoundAccepted, // 2
+  Reveal, // 3
 }
+export interface RPSData {
+  positionType: PositionType;
+  stake: BigNumber; // uint256
+  preCommit: string; // bytes32
+  playerAWeapon: Weapon;
+  playerBWeapon: Weapon;
+  salt: string; // bytes32
+}
+export interface Start {
+  type: 'start';
+}
+
+export type RoundProposed = Pick<RPSData, 'stake' | 'preCommit'> & {type: 'roundProposed'};
+
+export type RoundAccepted = Pick<RPSData, 'stake' | 'preCommit'> & {type: 'roundAccepted'};
```

Comment:
You need to add `bWeapon` to this. Might be worth re-writing as above too.

Reply:
The `salt` as well, I believe.